### PR TITLE
Add product related shipping costs to PayPalExpress payment service provider

### PR DESCRIPTION
**Pull request by nvindice**
For the bugfix, see #269 - PayPalExpress service provider now adds all product related shipping costs to the shipping service costs.

### CODE REVIEW

File: lib/mshoplib/src/MShop/Service/Provider/Payment/PayPalExpress.php [hunk 1]
```diff
@@ -727,8 +731,8 @@ protected function getOrderDetails( \Aimeos\MShop\Order\Item\Base\Iface $orderBa
 				foreach( $orderBase->getService( 'delivery' ) as $service )
 				{
 					$deliveryPrices = $this->addPrice( $deliveryPrices, $service->getPrice() );
-
-					$values['L_SHIPPINGOPTIONAMOUNT' . $lastPos] = number_format( $service->getPrice()->getCosts(), 2, '.', '' );
+
```
Comment (aimeos) [inline lib/mshoplib/src/MShop/Service/Provider/Payment/PayPalExpress.php:734]: There are several trailing tab. Can you please remove them?

Comment (nvindice) [reply lib/mshoplib/src/MShop/Service/Provider/Payment/PayPalExpress.php]: Sorry, fixed this.

**Comment by jonaseberle** [comment]
No, for me that is not fixing it.
The thing is that my services do not have a price cost, but a price value and and a tax rate.

**Comment by nvindice** [comment]
@jonaseberle How is that even possible? In the JQAdm service configuration menu, it is only possible to define costs and tax rate, not a price value...?

grafik

**Comment by jonaseberle** [comment]
I guess if that change works for costs(), it is fine. It makes sense to me. It tackles per-product service costs.

In my case the delivery price is adapted by a plugin (Similar to the `FreeShippingPlugin`). 

(I tried hard to make service costs behave fine with different tax rates and tax-paying/non-tax-paying customers. In the end only using value() worked for me with taxRate().)

**Review by jonaseberle — APPROVED** [review]

**Comment by aimeos** [comment]
Thanks a lot for the PR and your testing!

**Pull request merged**